components/ui/Dropdown.tsx

```diff
@@ -1,20 +1,92 @@
-import { useEffect, useState } from "react";
+import { CheckCircleIcon } from "@heroicons/react/outline";
+import * as DropdownMenuPrimitive from "@radix-ui/react-dropdown-menu";
+import { ComponentProps, forwardRef } from "react";
 
-export default function Dropdown(props) {
-  const [open, setOpen] = useState(false);
+export const Dropdown = DropdownMenuPrimitive.Root;
 
-  useEffect(() => {
-    document.addEventListener("keyup", (e) => {
-      if (e.key === "Escape") {
-        setOpen(false);
+type DropdownMenuTriggerProps = ComponentProps<typeof DropdownMenuPrimitive["Trigger"]>;
+export const DropdownMenuTrigger = forwardRef<HTMLButtonElement, DropdownMenuTriggerProps>(
+  ({ className = "", ...props }, forwardedRef) => (
+    <DropdownMenuPrimitive.Trigger
+      {...props}
+      className={
+        props.asChild
+          ? className
+          : `relative inline-flex items-center px-3 py-2 ml-2 text-sm font-medium text-gray-700 bg-transparent rounded-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-1 focus:bg-gray-100 focus:ring-neutral-500 group-hover:text-black ${className}`
       }
-    });
-  }, [open]);
-
-  return (
-    <div onClick={() => setOpen(!open)} {...props}>
-      {props.children[0]}
-      {open && props.children[1]}
-    </div>
-  );
-}
+      ref={forwardedRef}
+    />
+  )
+);
+DropdownMenuTrigger.displayName = "DropdownMenuTrigger";
+
+type DropdownMenuContentProps = ComponentProps<typeof DropdownMenuPrimitive["Content"]>;
+export const DropdownMenuContent = forwardRef<HTMLDivElement, DropdownMenuContentProps>(
+  ({ children, ...props }, forwardedRef) => {
+    return (
+      <DropdownMenuPrimitive.Content
+        {...props}
+        className="z-10 mt-1 text-sm origin-top-right bg-white rounded-sm shadow-lg w-44 ring-1 ring-black ring-opacity-5 focus:outline-none"
+        ref={forwardedRef}>
+        {children}
+        <DropdownMenuPrimitive.Arrow />
+      </DropdownMenuPrimitive.Content>
+    );
+  }
+);
+DropdownMenuContent.displayName = "DropdownMenuContent";
+
+type DropdownMenuLabelProps = ComponentProps<typeof DropdownMenuPrimitive["Label"]>;
+export const DropdownMenuLabel = (props: DropdownMenuLabelProps) => (
+  <DropdownMenuPrimitive.Label {...props} className="px-3 py-2 text-neutral-500" />
+);
+
+type DropdownMenuItemProps = ComponentProps<typeof DropdownMenuPrimitive["CheckboxItem"]>;
+export const DropdownMenuItem = forwardRef<HTMLDivElement, DropdownMenuItemProps>(
+  ({ className = "", ...props }, forwardedRef) => (
+    <DropdownMenuPrimitive.Item
+      className={`text-sm text-gray-700 hover:bg-gray-100 hover:text-gray-900 ${className}`}
+      {...props}
+      ref={forwardedRef}
+    />
```

Comment:
Minor change (not a blocker) - if you revert order of `className` and spread, you can skip `className`. 

```suggestion
  (props, forwardedRef) => (
    <DropdownMenuPrimitive.Item
      {...props}
      className={classNames(`text-sm text-gray-700 hover:bg-gray-100 hover:text-gray-900 ${className}`, props.className)}
      ref={forwardedRef}
    />
```